toolbox/src/SetReferences.cpp

```diff
@@ -239,47 +292,57 @@ bool SetReferences::initialize(BlockInformation* blockInfo)
 bool SetReferences::terminate(const BlockInformation* blockInfo)
 {
     using namespace yarp::dev;
-    bool ok = true;
 
     // Get the RobotInterface and the DoFs
     const auto robotInterface = getRobotInterface();
     const auto dofs = robotInterface->getConfiguration().getNumberOfDoFs();
 
     // Get the IControlMode2 interface
     IControlMode* icmd2 = nullptr;
-    ok = robotInterface->getInterface(icmd2);
-    if (!ok || !icmd2) {
+    if (!robotInterface->getInterface(icmd2) || !icmd2) {
         wbtError << "Failed to get the IControlMode2 interface.";
         return false;
     }
 
-    if (pImpl->controlModes.front() != VOCAB_CM_POSITION) {
-        // Set all the controlledJoints VOCAB_CM_POSITION
-        pImpl->controlModes.assign(dofs, VOCAB_CM_POSITION);
-
-        ok = icmd2->setControlModes(pImpl->controlModes.data());
-        if (!ok) {
-            wbtError << "Failed to set control mode.";
-            return false;
-        }
-    }
-    else { // In Position mode, restore the default reference speeds
+    // In Position mode restore the default reference speeds
+    if (pImpl->referenceType == impl::ReferenceType::JOINT_POSITION) {
         // Get the interface
         yarp::dev::IPositionControl* interface = nullptr;
-        ok = robotInterface->getInterface(interface);
-        if (!ok || !interface) {
+        if (!robotInterface->getInterface(interface) || !interface) {
             wbtError << "Failed to get IPositionControl interface.";
             return false;
         }
         // Restore default reference speeds
-        if (interface) {
-            ok = interface->setRefSpeeds(pImpl->defaultRefSpeed.data());
-            if (!ok) {
-                wbtError << "Failed to restore default reference speed.";
+        if (!interface->setRefSpeeds(pImpl->defaultRefSpeed.data())) {
+            wbtError << "Failed to restore default reference speed.";
+            return false;
+        }
+    }
+    // In MOTOR_BEMF or MOTOR_KTAU restore the default values
+    else if (pImpl->referenceType == impl::ReferenceType::MOTOR_BEMF
+             || pImpl->referenceType == impl::ReferenceType::MOTOR_KTAU) {
+        // Get the interface
+        ITorqueControl* interface = nullptr;
+        if (!robotInterface->getInterface(interface) || !interface) {
+            wbtError << "Failed to get ICurrentControl interface.";
+            return false;
+        }
+        for (unsigned m = 0; m < dofs; ++m) {
+            if (!interface->setMotorTorqueParams(m, pImpl->defaultMotorParams[m])) {
```

Comment:
See https://github.com/robotology/wb-toolbox/pull/124/commits/80273ce43ef0a0e7dab083772ced9f843bffc781#r191138043  but for the set (https://github.com/robotology/yarp/blob/devel/src/libYARP_dev/src/devices/RemoteControlBoard/RemoteControlBoard.cpp#L2597).